classes/security/authorization/internal/SubmissionFileAssignedQueryAccessPolicy.inc.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * @file classes/security/authorization/internal/SubmissionFileAssignedQueryAccessPolicy.inc.php
+ *
+ * Copyright (c) 2014-2015 Simon Fraser University Library
+ * Copyright (c) 2000-2015 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class SubmissionFileAssignedQueryAccessPolicy
+ * @ingroup security_authorization_internal
+ *
+ * @brief Submission file policy to check if the current user is a participant
+ * 	in a query the file belongs to.
+ *
+ */
+
+import('lib.pkp.classes.security.authorization.internal.SubmissionFileBaseAccessPolicy');
+
+class SubmissionFileAssignedQueryAccessPolicy extends SubmissionFileBaseAccessPolicy {
+	/**
+	 * Constructor
+	 * @param $request PKPRequest
+	 */
+	function SubmissionFileAssignedQueryAccessPolicy($request, $fileIdAndRevision = null) {
+		parent::SubmissionFileBaseAccessPolicy($request, $fileIdAndRevision);
+	}
+
+
+	//
+	// Implement template methods from AuthorizationPolicy
+	//
+	/**
+	 * @see AuthorizationPolicy::effect()
+	 */
+	function effect() {
+		$request = $this->getRequest();
+
+		// Get the user
+		$user = $request->getUser();
+		if (!is_a($user, 'PKPUser')) return AUTHORIZATION_DENY;
+
+		// Get the submission file
+		$submissionFile = $this->getSubmissionFile($request);
+		if (!is_a($submissionFile, 'SubmissionFile')) return AUTHORIZATION_DENY;
+
+		// Check if it's associated with a note.
+		if ($submissionFile->getAssocType() != ASSOC_TYPE_NOTE) return AUTHORIZATION_DENY;
+
+		$noteDao = DAORegistry::getDAO('NoteDAO');
+		$note = $noteDao->getById($submissionFile->getAssocId());
+		if (!is_a($note, 'Note')) return AUTHORIZATION_DENY;
+
+		if ($note->getAssocType() != ASSOC_TYPE_QUERY) return AUTHORIZATION_DENY;
+		$queryDao = DAORegistry::getDAO('QueryDAO');
+		$query = $queryDao->getById($note->getAssocId());
+		if (!$query) return AUTHORIZATION_DENY;
+
+		if (in_array($user->getId(), $queryDao->getParticipantIds($note->getAssocId()))) {
```

Comment:
why not pass the userId as a second parameter to limit the search and avoid in array?